app/packages/state/src/hooks/useStateUpdate.ts

```diff
@@ -52,6 +52,9 @@ const useStateUpdate = () => {
 
       if (state) {
         const view = get(viewAtoms.view);
+        if (dataset.stages && !state.view) {
+          state.view = dataset.stages;
```

Comment:
Does this mean that If dataset.stages exist, then we are loading a saved view? 

In that case, would it make sense or be possible to set state.view_name here?

Reply:
Isn't `viewName` already set from routing? Is there specific behavior you are thinking about?

Reply:
It wasn't previously. The view name is initially set in `set_view` and afterwards the url is updated in order to sync the other tabs.